var/spack/repos/builtin/packages/foam-extend/package.py

```diff
@@ -40,207 +130,234 @@ class FoamExtend(Package):
-    depends_on('metis@5:', when='+metis')
-    depends_on('parmetis', when='+parmetis')
-    depends_on('parmgridgen', when='+parmgridgen')
+    depends_on('scotch~metis',     when='~ptscotch+scotch')
+    depends_on('scotch~metis+mpi', when='+ptscotch')
+    depends_on('metis@5:',         when='+metis')
+    depends_on('parmetis',         when='+parmetis')
+    depends_on('parmgridgen',      when='+parmgridgen')
+    depends_on('paraview@:5.0.1',  when='+paraview')
+
+    # Some user settings, to be adjusted manually or via variants
+    foam_cfg = {
+        'WM_COMPILER':        'Gcc',  # <- %compiler
+        'WM_ARCH_OPTION':     '64',   # (32/64-bit on x86_64)
+        # FUTURE? 'WM_LABEL_SIZE':      '32',  # <- +int64
+        'WM_PRECISION_OPTION': 'DP',  # <- +float32
+        'WM_COMPILE_OPTION':  'SPACKOpt',  # Do not change
+        'WM_MPLIB':           'SPACK',     # Do not change
+    }
+
+    # The system description is frequently needed
+    foam_sys = {
+        'WM_ARCH':      None,
+        'WM_COMPILER':  None,
+        'WM_OPTIONS':   None,
+    }
+
+    build_script = './spack-Allwmake'  # <- Generated by patch() method.
+    # phases = ['configure', 'build', 'install']
 
-    depends_on('paraview', when='+paraview')
+    def setup_environment(self, spack_env, run_env):
+        run_env.set('FOAM_INST_DIR', self.prefix)
+        run_env.set('WM_PROJECT_DIR', self.projectdir)
+
+    @property
+    def _canonical(self):
+        """Canonical name for this package and version"""
+        return 'foam-extend-{0}'.format(self.version.up_to(2))
+
+    @property
+    def projectdir(self):
+        """Absolute location of project directory: WM_PROJECT_DIR/"""
+        return join_path(self.prefix, self._canonical)  # <- prefix/canonical
+
+    @property
+    def etc(self):
+        """Absolute location of the OpenFOAM etc/ directory"""
+        return join_path(self.projectdir, 'etc')
+
+    @property
+    def archbin(self):
+        """Relative location of architecture-specific executables"""
+        wm_options = self.set_openfoam()
+        return join_path('applications', 'bin', wm_options)
+
+    @property
+    def archlib(self):
+        """Relative location of architecture-specific libraries"""
+        wm_options = self.set_openfoam()
+        return join_path('lib', wm_options)
+
+    @property
+    def mplib_content(self):
+        """Define 'SPACK' mpi settings to have wmake
+        use spack information with minimum modifications to OpenFOAM
+        """
+        return [
+            'PFLAGS = -DOMPI_SKIP_MPICXX -DMPICH_IGNORE_CXX_SEEK',
+            'PINC   = -I{0}'.format(self.spec['mpi'].prefix.include),
+            'PLIBS  = -L{0} -lmpi'.format(pkglib(self.spec['mpi']))
+        ]
+
+    @property
+    def rpath_info(self):
+        """Define 'SPACKOpt' compiler optimization file to have wmake
+        use spack information with minimum modifications to OpenFOAM
```

Comment:
I don't know much about wmake -- do you need to manually set up RPATHs, or can wmake just rely on Spack's compiler wrappers adding the RPATHs?  Typically it's easier to do this if you can patch your make system to use Spack's custom compilers.